intbot/core/integrations/github.py

```diff
@@ -55,127 +77,175 @@ def as_discord_message(self):
 
-        return f"[@{login}]({url})"
+        if typename == "Issue":
+            return GithubIssue.model_validate(content)
+        elif typename == "DraftIssue":
+            return GithubDraftIssue.model_validate(content)
+        else:
+            raise ValueError("Other types are not supported")
 
-    def content_type(self):
-        return self.content["projects_v2_item"]["content_type"]
+    def get_project(self) -> GithubProject:
+        return GithubProject.model_validate(self.extra["project"])
 
-    def node_id(self):
-        # NOTE(artcz): This is relevant, because of how the graphql query above
-        # is constructed.
-        # Using node_id, which is an id of a ProjectV2Item we can get both
-        # DraftIssue and Issue from one query.
-        # If we use the content_node_id we would probably need two separate
-        # ways of getting that data.
-        return self.content["projects_v2_item"]["node_id"]
+    def get_repository(self):
+        # Not relevant at the moment
+        return ...
 
     def changes(self) -> dict:
-        if "changes" in self.content:
-            fv = self.content["changes"]["field_value"]
-            field_name = fv["field_name"]
-            field_type = fv["field_type"]
-
-            if field_type == "date":
-                changed_from = (
-                    fv["from"].split("T")[0] if fv["from"] is not None else "None"
-                )
-                changed_to = fv["to"].split("T")[0] if fv["to"] is not None else "None"
-
-            elif field_type == "single_select":
-                changed_from = fv["from"]["name"] if fv["from"] is not None else "None"
-                changed_to = fv["to"]["name"] if fv["to"] is not None else "None"
-
-            else:
-                changed_from = "None"
-                changed_to = "None"
-
-            return {
-                "field": field_name,
-                "from": changed_from,
-                "to": changed_to,
-            }
 
-        return {}
+        # Early return! \o/
+        if "changes" not in self.content:
+            # Fallback because some webhooks just don't have changes.
+            return {}
 
-    def as_discord_message(self, github_object: GithubDraftIssue | GithubIssue) -> str:
-        message = "{sender} {action} {details}".format
+        fv = self.content["changes"]["field_value"]
+        field_name = fv["field_name"]
+        field_type = fv["field_type"]
+
+        if field_type == "date":
+            changed_from = (
+                fv["from"].split("T")[0] if fv["from"] is not None else "None"
+            )
+            changed_to = fv["to"].split("T")[0] if fv["to"] is not None else "None"
+
+        elif field_type == "single_select":
+            changed_from = fv["from"]["name"] if fv["from"] is not None else "None"
+            changed_to = fv["to"]["name"] if fv["to"] is not None else "None"
+
+        else:
+            changed_from = "None"
```

Comment:
Soo it's a date, but its representation is a string and if there's no date, then we use `None` as a string? It's a bit confusing what type `changed_from` is - if this is the intention, please add a type hint to the line 162 and 165.

Reply:
Added type hints to clarify